pkg/inclusion/nmt_caching_test.go

```diff
@@ -128,8 +128,10 @@ func TestEDSSubRootCacher(t *testing.T) {
 	for i := range dah.RowRoots[:squareSize] {
 		expectedSubTreeRoots := calculateSubTreeRoots(t, eds.Row(uint(i))[:squareSize], 2)
 		require.NotNil(t, expectedSubTreeRoots)
-		// note: the depth is one greater than expected because we're dividing
-		// the row in half when we calculate the expected roots.
+		// note: the walk path has 3 elements (depth 3) while calculateSubTreeRoots
+		// uses depth 2, so we're comparing subtree roots at different depths.
+		// The walk path [false, false, false] corresponds to the leftmost subtree
+		// at depth 3, which should match the first subtree root calculated at depth 2.
```

Comment:
I don't understand the comment before or after this change.

Instead of this change, can you please refactor the test so that it's less confusing? A few ideas that would help me:

1. It is confusing that `NewSubtreeCacher` returns a type `EDSSubTreeRootCacher`. We should use consistent names so instead it should be: `NewSubtreeCacher` returns a `SubtreeCacher`

2. Can the test `calculateSubTreeRoots` and `getSubTreeRoot` use the same depth?